badge-maker/lib/make-badge.js

```diff
@@ -23,6 +23,10 @@ module.exports = function makeBadge({
   logoWidth,
   links = ['', ''],
 }) {
+  if (!logo && (logoPosition !== undefined || logoWidth !== undefined)) {
+    throw Error('`logoPosition` and `logoWidth` require `logo`')
+  }
+
```

Comment:
I feel like we've been here before. I can't remember what param it was (probably a different one), but we added some "if you're specifying this param, it has to actually make sense" logic before, loads of stuff broke and we ended up rolling it back the next day. Does that ring a bell.

Reply:
> "wants" is a strong word 😉

😝 

Hmm, no… it doesn't quite ring a bell. Unless it was something in the endpoint badge? I wonder if we added a test. I can do a little digging in the issues.

The reason I added this is so I could remove what appeared to be logic guarding against this inside the render functions.

I think, probably, that if there's a good reason for the server to accept params that don't make sense, that it should strip them out before invoking `makeBadge()`.

Reply:
I can't remember for the life of me what it is now, but we did something vaguely sensible-souning like like

- `?style=` has to be an actual style
- if you're going to set `?logoColor` you have to set `?logo=`
- `?color=` has to be an actual colour

(it might have been none of those). It seemed completely valid at the time, then we deployed it and realised loads of people had badges out there with spurious values in the URLs and had to roll it back.

Spent about an hour trawling through old issues and PRs to try and track it down - can't find it :(

Reply:
I guess the question is what happens when you pass such a URL to the badge server, and does it even get as far as these renderers? How about I move this change to its own draft PR and we can add some tests and investigate. As the code stands now, I don't think it is critical anymore.